packages/editor/src/browser/editor-quick-open-service.ts

```diff
@@ -0,0 +1,154 @@
+    protected readonly workspaceService: WorkspaceService;
+
+    readonly prefix: string = 'edt ';
+
+    get description(): string {
+        return 'Show All Opened Editors';
+    }
+
+    getModel(): QuickOpenModel {
+        return this;
+    }
+
+    getOptions(): QuickOpenOptions {
+        return {
+            placeholder: '',
+            fuzzyMatchLabel: {
+                enableSeparateSubstringMatching: true
+            },
+            fuzzyMatchDescription: {
+                enableSeparateSubstringMatching: true
+            },
+            onClose: () => {
+                this.cancelIndicator.cancel();
+            }
+        };
+    }
+
+    open(): void {
+        this.prefixQuickOpenService.open(this.prefix);
+    }
+
+    private cancelIndicator = new CancellationTokenSource();
+
+    public async onType(lookFor: string, acceptor: (items: QuickOpenItem[]) => void): Promise<void> {
+        this.cancelIndicator.cancel();
+        this.cancelIndicator = new CancellationTokenSource();
+        const token = this.cancelIndicator.token;
+
+        const editorItems: QuickOpenItem[] = [];
+
+        // Get URIs of all EditorWidgets that are currently opened, sorted by URI
+        const widgets: URI[] = this.editorManager.all
+            .filter((w: EditorWidget) => !!w.getResourceUri())
+            .map((w: EditorWidget) => w.getResourceUri()!)
+            .sort();
+
+        if (widgets.length === 0) {
+            editorItems.push(new QuickOpenItem({
+                label: 'List of opened editors is currently empty',
+                run: () => false
+            }));
+            acceptor(editorItems);
+            return;
+        }
+
+        for (const uri of widgets) {
+            const item = await this.toItem(uri);
+            if (!!token.isCancellationRequested) {
+                return;
+            }
+            editorItems.push(item);
+            acceptor(editorItems);
+        }
+        return;
+    }
+
+    private async toItem(uri: URI, group?: QuickOpenGroupItemOptions): Promise<QuickOpenItem<QuickOpenItemOptions>> {
+        let description = this.labelProvider.getLongName(uri.parent);
+        if (this.workspaceService.workspace && !this.workspaceService.workspace.isDirectory) {
+            const rootUri = this.workspaceService.getWorkspaceRootUri(uri);
+            if (rootUri) {
+                description = `${rootUri.displayName} • ${description}`;
+            }
+        }
+
+        const options: QuickOpenItemOptions = {
+            label: this.labelProvider.getName(uri),
+            iconClass: await this.labelProvider.getIcon(uri) + ' file-icon',
+            description: description,
+            tooltip: uri.path.toString(),
+            uri: uri,
+            hidden: false,
+            run: this.getRunFunction(uri)
+        };
+        return new QuickOpenItem<QuickOpenItemOptions>(options);
+    }
+
+    protected getRunFunction(uri: URI): (mode: QuickOpenMode) => boolean {
```

Comment:
It might be good to provide documentation for the following method (ex: it calls `openFile`), else we can simply refactor and use the implementation directly when specifying the `run` in `toItem`.